Lib/functools.py

```diff
@@ -901,6 +901,15 @@ def __init__(self, func):
         self.dispatcher = singledispatch(func)
         self.func = func
 
+        # bpo-45678: special-casing for classmethod/staticmethod in Python <=3.9,
+        # as functools.update_wrapper doesn't work properly in singledispatchmethod.__get__
+        # if it is applied to an unbound classmethod/staticmethod
+        self._wrapped_func = (
+            func.__func__
+            if isinstance(func, (staticmethod, classmethod))
+            else func
+        )
+
```

Comment:
Ironically the crude spelling is in this case shorter and more obvious:

```suggestion
        if isinstance(func, (staticmethod, classmethod)):
            self._wrapped_func = func.__func__
        else:
            self._wrapped_func = func
```